apm-sniffer/apm-toolkit-activation/apm-toolkit-opentracing-activation/src/main/java/org/skywalking/apm/toolkit/activation/opentracing/span/ConstructorWithSpanBuilderInterceptor.java

```diff
@@ -0,0 +1,42 @@
+package org.skywalking.apm.toolkit.activation.opentracing.span;
+
+import org.skywalking.apm.agent.core.context.ContextCarrier;
+import org.skywalking.apm.agent.core.context.ContextManager;
+import org.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.skywalking.apm.toolkit.opentracing.SkywalkingSpanBuilder;
+import org.skywalking.apm.toolkit.opentracing.Tag;
+import org.skywalking.apm.util.StringUtil;
+
+public class ConstructorWithSpanBuilderInterceptor implements InstanceConstructorInterceptor {
+
+    @Override
+    public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
+        SkywalkingSpanBuilder spanBuilder = (SkywalkingSpanBuilder)allArguments[0];
+
+        if (spanBuilder.isEntry()) {
+            ContextManager.createEntrySpan(spanBuilder.getOperationName(), null);
+        } else if (spanBuilder.isExit() && (!StringUtil.isEmpty(spanBuilder.getPeer()))) {
+            ContextManager.createExitSpan(spanBuilder.getOperationName(),
+                new ContextCarrier(), buildRemotePeer(spanBuilder));
+        } else {
+            ContextManager.createLocalSpan(spanBuilder.getOperationName());
+        }
+
+        AbstractSpan span = ContextManager.activeSpan();
```

Comment:
No need to use `activeSpan()` method, `create*Span` method returns the `AbstractSpan`.